awsx/ec2/subnetDistributor.ts

```diff
@@ -126,17 +156,22 @@ export function getSubnetSpecs(
         splitBase = azBases[i];
       }
 
-      const isolatedCidrMask: number = isolatedSubnetsIn[j].cidrMask ?? 24;
+      const isolatedCidrMask: number =
+        isolatedSubnetsIn[j].cidrMask ?? Math.max(baseSubnetMask + newBitsPerSubnet, 24);
       const newBits = isolatedCidrMask - baseIp.subnetMask;
+      let nextAddress = new ipAddress.Address4(cidrSubnetV4(splitBase, newBits, j));
+      if (isOverlapping(currentAddress, nextAddress)) {
+        nextAddress = nextBlock(currentAddress, isolatedCidrMask);
```

Comment:
Shouldn't this increment nextAddress, not currentAddress? Also is it guaranteed that the new nextAddress does not overlap with currentAddress?

Reply:
Yup, it does - it firstly overwrites the nextAddress because the old nextAddress was overlapping, then on line 174 we allocate the nextAddress back to the currentAddress. I've used this layout in each looop.

Yes, the method used in nextBlock is guarenteed to not overlap.